boa_engine/src/syntax/ast/node/visit.rs

```diff
@@ -0,0 +1,2317 @@
+use crate::syntax::ast::node::declaration::class_decl::ClassElement;
+use crate::syntax::ast::node::declaration::{
+    BindingPatternTypeArray, BindingPatternTypeObject, DeclarationPatternArray,
+    DeclarationPatternObject,
+};
+use crate::syntax::ast::node::iteration::IterableLoopInitializer;
+use crate::syntax::ast::node::object::{MethodDefinition, PropertyDefinition, PropertyName};
+use crate::syntax::ast::node::operator::assign::AssignTarget;
+use crate::syntax::ast::node::template::TemplateElement;
+use crate::syntax::ast::node::{
+    ArrayDecl, ArrowFunctionDecl, Assign, AsyncFunctionDecl, AsyncFunctionExpr, AsyncGeneratorDecl,
+    AsyncGeneratorExpr, AwaitExpr, BinOp, Block, Break, Call, Case, Catch, Class, ConditionalOp,
+    Continue, Declaration, DeclarationList, DeclarationPattern, DoWhileLoop, Finally, ForInLoop,
+    ForLoop, ForOfLoop, FormalParameter, FormalParameterList, FormalParameterListFlags,
+    FunctionDecl, FunctionExpr, GeneratorDecl, GeneratorExpr, GetConstField, GetField,
+    GetPrivateField, GetSuperField, Identifier, If, New, Object, Return, Spread, StatementList,
+    SuperCall, Switch, TaggedTemplate, TemplateLit, Throw, Try, UnaryOp, WhileLoop, Yield,
+};
+use crate::syntax::ast::{op, Const, Node};
+use boa_interner::Sym;
+
+/// `Visitor`s "walk" the AST at certain nodes. Useful for when you need to modify source code, but at
+/// the API level.
+///
+/// `Visitor`s have default implementations which are "reasonable", which is to say that users merely
+/// need to implement the visitor for the specific type they wish to visit. For example, let's say
+/// we want to replace all variables named "y" with a variable named "z".
+///
+/// To do so, we implement `Visitor` over a `NameReplacer` struct, then use `NameReplacer` to, well,
+/// replace the names in a parsed source.
+///
+/// ```
+/// use boa_engine::syntax::ast::node::visit::Visitor;
+/// use boa_engine::Context;
```

Comment:
We usually prefer scoped imports:
```suggestion
/// use boa_engine::{Context, syntax::ast::node::visit::Visitor};
```